### PR TITLE
refactor ~ improve code coverage implementation

### PR DESCRIPTION
**Pull request by rivy**
* modifies GHA:CICD workflow to successfully calculate code coverage for macos, unix, and windows
* uploads coverage maps to CodeCov.io (which can integrate reports from multiple platforms)
* includes shortcut utilities to calculate and show locally generated code coverage

### CODE REVIEW

**Comment by rivy** [comment]
@sylvestre , if you have some time, would you look at the "util/show-codecov.sh" shell script?

Specifically, I've marked three lines "FixME". Probably just a brain misfire, but I can't figure out how to code this to condense the ignore's for `grcov` into a variable and use it at the later two spots as an option to `grcov`. When I've tried to debug using `echo ...`, the command line seems correct, but `grcov` ignores the options included via `${GRCOV_IGNORE_OPTION}`. And, so, I'm having to duplicate the ignores, writing them directly on the commands lines.
😕

**Comment by rivy** [comment]
As a reminder, I'll update the GHA:CICD workflow to include some appropriate `grcov` line and branch exclusions (eg, `--excl-br-line 'assert(_eq|_ne)?!'`) when https://github.com/mozilla/grcov/pull/416 is merged.

File: src/factor/src/factor.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
+#![crate_name = "uu_factor"]
```
[the remark this answers is not in the document]
Comment (rivy) [reply src/factor/src/factor.rs:1]: Yeah, thanks for seeing that ... it looks like I missed some files when I reorganized.

I'm taking a fresh look at it this morning, and I think that I now have a better organization in mind. I'm currently reworking the PR commits and I'll push a fresh set this afternoon.

[the remark this answers is not in the document]
Comment (rivy) [reply src/factor/src/factor.rs]: Improved / fixed.

**Comment by sylvestre** [comment]
Yeah, bravo, thanks :)

**Pull request merged**